static/js/settings_profile_fields.js

```diff
@@ -231,7 +247,7 @@ export function parse_field_choices_from_field_data(field_data) {
             order: choice.order,
         });
     }
-
+    choices.sort((a, b) => a.order - b.order);
```

Comment:
We need this because currently the choices are returned from server in the ascending order of values/id of the option. We can skip this if we remove order field and instead pass a list as suggested in https://chat.zulip.org/#narrow/stream/378-api-design/topic/custom.20profile.20fields/near/1382989.